### ISSUE TITLE
Are you sure .train() puts batch norm in eval mode? I'd assume that means train...

### ISSUE THREAD

**Issue by brando90** [awslabs/aws-cv-task2vec#23]
https://github.com/awslabs/aws-cv-task2vec/blob/c5795e55ba773f9845498091a90eee2fcba5da31/task2vec.py#L71

```python
        # Fix batch norm running statistics (i.e., put batch_norm layers in eval mode)
        self.model.train()
```

is this truly correct?

.train() usually puts layers in training mode. So for batch norm what it means is that it start collecting running statistics but uses mini-batch stats if I remember correctly, while in .eval() it uses the saved running stats. Right?

**Comment by brando90** [comment]
# Since we are fine-tuning the model during for T2V/FIM computation, .train() is the right choice in general.
        self.model.train()

**Comment by brando90** [comment]
For batch norm comment:
```
# Since we are fine-tuning the model during T2V/FIM computation, .train() is the right choice as it ensures batch norm uses mini-batch statistics and properly adapts the model to the new task.
self.model.train()
```

but LLMs don't really use batch norm so doesn't matter...
